### ISSUE TITLE
Not compatible with latest version of docker

### ISSUE THREAD

**Issue by OpportunityLiu** [prebuild/prebuildify-cross#20]
Since docker api versions before v1.24 are [deprecated](https://docs.docker.com/engine/deprecated/#deprecate-legacy-api-versions). And the support for these version is removed in v26.0. prebuildfy-cross will not work with latest github actions images.

You will get
```
Error: {"message":"client version 1.21 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version"}
```

For reference:
https://docs.docker.com/engine/api/#deprecated-api-versions
https://github.com/actions/runner-images/pull/9982
https://github.com/CloudPSS/mmap/actions/runs/9662063371/job/26651153004

**Comment by vweevers** [comment]
We might get lucky if we just bump the version [here](https://github.com/vweevers/docker-pull/blob/2cca4eea5afb282791835946b3a183406b3b5fe3/index.js#L15) and if the subset of the API that we use did not have breaking changes.

Otherwise, we'll have to go trough https://github.com/moby/moby/blob/master/docs/api/version-history.md and change code accordingly.

**Comment by vweevers** [comment]
At quick glance, 1.24 is compatible with the API that we use. Could you try that out?

**Comment by OpportunityLiu** [comment]
The dependency `docker-run` hardcoded api version and not updated for 8 years.

https://github.com/mafintosh/docker-run/blob/9c685de188eac9c1a70cad9f0f7d22f5cd92aedc/index.js#L17

As of `@vweevers/docker-pull`, it accepts a `version` option.

https://github.com/vweevers/docker-pull?tab=readme-ov-file#var-p--pullimage-options-callback

Should we move to `dockerode`?

**Comment by vweevers** [comment]
> As of @vweevers/docker-pull, it accepts a version option.

Yes, but I prefer to change the version there, so both modules benefit. Either way, we need someone to confirm that it works.

> Should we move to `dockerode`?

No. There's no need to complicate things when all we need is replacing a "1.21" string with "1.24".

**Comment by OpportunityLiu** [comment]
> The dependency `docker-run` hardcoded api version and not updated for 8 years.
> 
> https://github.com/mafintosh/docker-run/blob/9c685de188eac9c1a70cad9f0f7d22f5cd92aedc/index.js#L17

So you have to make a fork of this package too. It seems I can't help.

**Comment by thegecko** [comment]
Please see https://github.com/prebuild/prebuildify-cross/pull/21

**Comment by vweevers** [comment]
I'll go for #21 because it fixes both `docker-pull` and `docker-run` while #22 only fixes `docker-pull`. Thank you both!

**Comment by vweevers** [comment]
Fixed in 5.1.1.

**Comment by fanatid** [comment]
Thank you @vweevers! :rocket:
